pkg/kubelet/dockertools/manager.go

```diff
@@ -1170,6 +1171,9 @@ func (dm *DockerManager) killContainer(containerID types.UID) error {
 					break
 				}
 			}
+			if pod.Spec.TerminationGracePeriodSeconds != nil {
+				gracePeriod = uint(*pod.Spec.TerminationGracePeriodSeconds)
```

Comment:
> On May 22, 2015, at 3:45 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/kubelet/dockertools/manager.go:
> 
> > @@ -1170,6 +1171,9 @@ func (dm *DockerManager) killContainer(containerID types.UID) error {
> >  					break
> >  				}
> >  			}
> > +			if pod.Spec.TerminationGracePeriodSeconds != nil {
> > +				gracePeriod = uint(*pod.Spec.TerminationGracePeriodSeconds)
> If we don't want to drive decisions off DeletionTimestamp due to potential clock skew, then we need to record the value of DeleteOptions.GracePeriodSeconds and use that.
> 
I'm doing that into pod.Spec.TerminationGracePeriodSeconds - is that not sufficient?  It seems like that's the proper place - we record the users intent to set a custom termination grace period (with their delete) into the spec.
> —
> Reply to this email directly or view it on GitHub.
>